docs/messaging.rst

```diff
@@ -1213,6 +1239,10 @@ it shall publish a status message with ``execution_state: 'idle'``.
 Thus, the outputs associated with a given execution shall generally arrive
 between the busy and idle status messages associated with a given request.
 
+A kernel may send optional status messages with execution states other than
+`busy` or `idle`. For example, a kernel may send a status message with a
+`starting` execution state exactly once at process startup.
+
```

Comment:
@minrk, do we know of other kernels publishing other status states? We made the 'starting' status optional because there *might* be a race condition preventing the client from ever seeing a starting message. However, it is nice if we have a universe of only three status states, rather than opening it up to any status the kernel might want to send.

As for the kernel status states that the notebook spoofs, we already agreed that those should come on a different channel, a notebook channel, so I think those status states shouldn't count here as optional states sent from a kernel.

Reply:
To be clear, I think I'm +1 on reverting this "optional status states" change, and just going back to mandating a 'starting' status, and having a defined universe of 'starting', 'busy', and 'idle' states sent from the kernel.

Reply:
I don't have an opinion on this.